diff --git a/src/outline.js b/src/outline.js
index f55e49b4ff02352ecf1352368b2ea06503434fcc..df30dbb7144c7c38eae0e8aef79e653fccc870c4 100644
--- a/src/outline.js
+++ b/src/outline.js
@@ -13,7 +13,7 @@ function sanitize (string) {
 }
 
 async function parseOutline(page, tags, enableWarnings) {
-	return await page.evaluate((tags) => {
+	return tags.length && await page.evaluate((tags) => {
 		const tagsToProcess = [];
 		for (const node of document.querySelectorAll(tags.join(","))) {
 			tagsToProcess.push(node);
diff --git a/src/printer.js b/src/printer.js
index 206b4fcad71822e5c92b36fb469346a4cdd56bfb..d20e294afa9e536c58e808475deff4c80ceac450 100644
--- a/src/printer.js
+++ b/src/printer.js
@@ -324,7 +324,7 @@ class Printer extends EventEmitter {
 
 			setMetadata(pdfDoc, meta);
 			setTrimBoxes(pdfDoc, this.pages);
-			setOutline(pdfDoc, outline, this.enableWarnings);
+			outline && setOutline(pdfDoc, outline, this.enableWarnings);
 
 			pdf = await pdfDoc.save();
 
